pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ScanBasedDocIdIterator.java

```diff
@@ -24,22 +24,18 @@
 
 
 /**
- * All scan based filter iterators must implement this interface. This allows intersection to be
- * optimized.
- * For example, if the we have two iterators one index based and another scan based, instead of
```

Comment:
This is good piece of information. Why delete it? That's how the filtering will work right if we do 
WHERE col1 = 200 AND col2 = 10 -- if there is an inverted index on col1 and no index on col2

Reply:
This part is explained in the AndDocIdSet. Updated the javadoc so that it is more clear here as well.